README.md

```diff
@@ -15,102 +13,350 @@ For a holistic guide to using Nemo as an overall automation solution, [please st
+* **wd** This is a reference to the `selenium-webdriver` module: http://selenium.googlecode.com/git/docs/api/javascript/module_selenium-webdriver.html
 
-In addition to the Nemo setup using the name/value pairs, the NVPs are also passed along to the nemo object returned after setup,
-in the "props" namespace. You also can pass in arbitrary name/values through `nemoData`. Sometimes you might find it useful to have
-such data passed in and used in your test scripts. Also, some plugins may require certain name/values to be defined here.
+* **data** Pass through of any data included in the `data` property of the configuration
 
-Note the use in the example of the arbitrary NVP "targetBaseUrl" and how it is accessed later on in the script as `nemo.props.targetBaseUrl`
+* **_config** This is the confit configuration object. Documented here: https://github.com/krakenjs/confit#config-api
 
-Here are the required variables and their meaning:
+* **plugin namespace(s)** Plugins are responsible for registering their own namespace under nemo. The convention is that the plugin should register the same namespace
+as is specified in the plugin configuration. E.g. the `nemo-view` module registers itself as `nemo.view` and is configured like this:
 
-### autoBaseDir
+```javascript
+{
+  "driver": ...
+  "plugins": {
+    "view": {
+      "module": "nemo-view"
+    }
+  }
+```
+
+You could also have a config that looks like this, and `nemo-view` will still register itself as `nemo.view`
+
+```javascript
+{
+  "driver": ...
+  "plugins": {
+    "cupcakes": {
+      "module": "nemo-view"
+    }
+  }
+```
+
+But that's confusing. So please stick to the convention.
+
+### Typical usage of Nemo constructor
+
+A typical pattern would be to use `mocha` as a test runner, resolve `nemo` in the context of the mocha `before` function, and use
+the mocha `done` function as the callback:
+
+```javascript
+var nemo;
+describe('my nemo suite', function() {
+  before(function(done) {
+    nemo = Nemo(config, done);
+  });
+  it('will launch browsers!', function(done) {
+    nemo.driver.get('https://www.paypal.com');
+    nemo.driver.quit().then(function() {
+       done();
+    });
+  });
+});
+
+```
+
+## Nemo Configuration
+
+```javascript
+{
+  "driver": { /** properties used by Nemo to setup the driver instance **/ },
+  "plugins": { /** plugins to initialize **/},
+  "data": { /** arbitrary data to pass through to nemo instance **/ }
+}
+```
+
+This configuration object is optional, as long as you've got `nemoData` set as an environment variable (see below).
 
-Used as a root location for finding 'locator' files in your test suite. Also can be required by other plugins
-which need to require modules from your test suite.
+### driver
 
-### targetBrowser
+Here are the `driver` properties recognized by Nemo. This is ALL of them. Please be aware that you really only need to supply "browser" to get things working initially.
+
+#### browser (optional)
```

Comment:
It looks like browser is not optional. we don't default it to anything and that makes driver not initialize. For example below example doesn't work locally 
```
var nemo = Nemo({
    "driver": {
    },
    'data': {
        'baseUrl': 'https://www.paypal.com'
    }
}, function (err) {
    //always check for errors!
    if (!!err) {
        console.log('Error during Nemo setup', err);
    }
    nemo.driver.get(nemo.data.baseUrl);
    nemo.driver.getCapabilities().
        then(function (caps) {
            console.info("Nemo successfully launched-", caps.caps_.browserName);
        });
    nemo.driver.quit();
});```